### DISCUSSION TITLE
Understanding how to access the live-stream from other devices on the same network

### DISCUSSION THREAD

**Discussion by roshanyadav29** [signag/raspi-cam-srv#24] [Q&A] [answered]
Hi, first of all, Thank you very much, Raspi-cam-srv is working fabulously. The documentation is great, even as someone who has practically zero idea about flask and web hosting, I was able to get it up and running.
I wanted to understand how I could proceed with accessing the live-stream from any other device. Is it even possible?

**Comment by signag** [comment]
Hi,

if you open the *Web Cam* page ([Web Cam](https://github.com/signag/raspi-cam-srv/blob/main/docs/Webcam.md)), you will see the URLs for accessing live streams and photo for the available cameras.

On the [Settings](https://github.com/signag/raspi-cam-srv/blob/main/docs/Settings.md) screen, you can specify whether external streaming through these end points requires authentication or not.

Hope this answers your question.

**Reply by roshanyadav29** [reply]
Awesome, I had tried the authentication part. It seemed like a neat feature. Unfortunately, I can't seem to access the video_feed from any other device even though I am on the same wifi as RPi 4b. I tried this address- 127.0.0.1:5000/video_feed. It works on the Rpi but not on another device. Am I doing something wrong here?

**Reply by roshanyadav29** [reply]
Update: After some digging, It seems like it was a simple problem. According to what I can understand, Flask is made for development and not for production. Hence the local address 127.0.0.1 is only accessible in rpi. A way around it is just to use your Ipv4 (available if you hover over wifi symbol in rpi) instead of default 0.0.0.0. Then you can go to that with other device along with port and it will work.

I will create a sepearate discussion for other queries. Thank you for helping. Again, Great work with the Repo.

**Comment by signag** [comment]
It's strange that you tried localhost(127.0.0.1).

The *Web Cam* page is programmed to show links with the **hostname** of your Raspi.    
With this, you should be able to reach it from everywhere (provded, there is a DNS server in your network).

![image](https://github.com/user-attachments/assets/2e749850-11d6-4e47-878b-36a2c8918a56)

What does your screen show?

Or did you not set a hostname when setting up your Raspi?

hostname and port should also be visible in the upper right corner.

**Reply by roshanyadav29** [reply]
Top right corner showed exactly that 127.0.0.1:5000, did I miss a step somewhere? and the links were the same with /video_feed after them. How can I go about changing my hostname? Thanks

**Comment by signag** [comment]
Normally, you specify the hostname when setting up an SD card with Imager (see [OS Customization](https://www.raspberrypi.com/documentation/computers/getting-started.html#advanced-options)).

In Linux, you can show the currently set hostname with one of the commands    
```hostnamectl status```    
```hostnamectl hostname```    
```hostnamectl```    
```hostname```

You set/change the hostname with    
```sudo hostnamectl hostname NEWNAME```    
where you need to replace ```NEWNAME``` with your intended hostname.

The new hostname will only be visible in your network after you reboot the Pi:   
```sudo reboot```

Please let me know whether or not a hostname was set in your system, so that I can put a note into the documentation.

**Reply by roshanyadav29** [reply]
I had setup the hostname for the rpi as admin (all small case) along with a password while installing rpi os in the sd card. Not sure why, it was showing 127.0.0.1

**Comment by signag** [comment]
Are you starting Flask with    
```flask --app raspiCamSrv run --port 5000 --host=0.0.0.0```    
as described in step 13 of the [Installation](https://github.com/signag/raspi-cam-srv#raspicamsrv-installation) guide?

The ```--host=0.0.0.0``` is important. It assures that the Flask Server can be reached from other computers.    
The default for ```host``` is 127.0.0.1.

See **EXTERNALLY VISIBLE SERVER** in the [Flask Documentation](https://flask.palletsprojects.com/en/3.0.x/quickstart/#a-minimal-application)

**Reply by roshanyadav29** [reply]
I stoped the previously running service just in case and re-did it with --host=0.0.0.0. But I can see 127.0.0.1:5000 and 192.168.1.2:5000 (I tried using this to see if I could access it with another device) as running. But on both of them the Streaming URL is //localaddress:port/video_feed. I have attached an image for reference.

![rpi_cam](https://github.com/user-attachments/assets/7f0c89b9-25ff-436f-bcf4-f9b6f6f83f8d)

**Comment by signag** [comment, marked as answer]
As far as I can see from your screenshot:   
- The IP address of your Raspberry Pi is ```192.168.1.2```
- The hostname of your Raspberry Pi is ```raspberrypi```
- The user logged in is ```admin```

Can you please confirm with command    
```hostnamectl hostname```   
that the hostname is ```raspberrypi```

Now you can try from another computer in your network    
1. ```ping 192.168.1.2```
2. ```ping raspberrypi```

If the second ping with the hostname does not work, you may have an issue with name resolution in your network.    
Perhaps there is no DNS server configured.     
You can check this on Linux with    
```cat /etc/resolv.conf```    
or on Windows with    
```ipconfig /all```

The 3 lines after the red warning after start of Flask tell you that the Flask server is listening on all addresses.     

So, when starting a browser on the Raspberry Pi itself, you should be able to stream with  
```http://127.0.0.1:5000/video_feed```    
which is identical to    
```http:localhost:5000/video_feed```

When startíng a browser on another computer, you should be able to stream with  
```http://192.168.1.2:5000/video_feed```     
Of course, you cannot use ```172.0.0.1``` here, because this refers to the address of the computer where the browser is running, which is obviously not the one where Flask is running.

With a browser on another computer, you might also be able to stream with    
```http://raspberrypi:5000/video_feed```     
if the 2. ping from above has been working.

Using host names instead of IP addresses has the advantage that they do not change in case that your DHCP server assings a different IP address.

**Reply by roshanyadav29** [reply]
Awesome, this resolved all of my concerns. You are correct about my hostname being raspberrypi, (sorry I misunderstood earlier). And It is working great with using hostname instead of the IP. Thank you for your help in this.
